### PR TITLE
svcenc: RC bitrate overflows prevented

### PR DESCRIPTION
**Pull request by AshwinNatesan-ittiam**
RC uses int32_t for storing multiple quantities related to bits in a given period.
'isvce_svc_frame_params_validate' has been added, which queries RC API for the relevant quantities and returns with and error if computations involving those quantities exceed INT32_MAX.

BUG = ossfuzz:274221347
BUG = ossfuzz:274265498
Test: svc_enc_fuzzer

### CODE REVIEW

File: fuzzer/svc_enc_fuzzer.cpp [hunk 1]
```diff
@@ -545,7 +545,7 @@ bool Codec::initEncoder(const UWORD8 *data)
     isvce_init_ip_t sInitIp{};
     isvce_init_op_t sInitOp{};
 
-    std::vector<UWORD32> sMaxBitrates(mNumSpatialLayers, 240000000);
+    std::vector<UWORD32> sMaxBitrates(mNumSpatialLayers, 6000000);
```
Comment (harishdm) [inline fuzzer/svc_enc_fuzzer.cpp:548]: is this change really necessary? does this mean max bitrate for each layer is 6mbps? isn't that low?

Comment (AshwinNatesan-ittiam) [reply fuzzer/svc_enc_fuzzer.cpp:548]: This is not needed. I will revert it.

**Review by harishdm — APPROVED** [review]
